integration_tests/src/bin/benchmark/benchmark_pass.rs

```diff
@@ -0,0 +1,340 @@
+            wgpu_ctx: Some(ctx),
+            ..benchmark_pipeline_options(self.framerate)
+        })?;
+
+        let pipeline = Arc::new(Mutex::new(pipeline));
+
+        debug!("pipeline start");
+        Pipeline::start(&pipeline);
+
+        let inputs: Vec<_> = (0..self.input_count)
+            .map(|i| InputId(format!("input_{i}").into()))
+            .collect();
+        let outputs: Vec<_> = (0..self.output_count)
+            .map(|i| OutputId(format!("output_{i}").into()))
+            .collect();
+
+        match self.input_file.clone() {
+            InputFile::Raw(input) => {
+                let frame_senders = inputs
+                    .iter()
+                    .cloned()
+                    .map(|input_id| Ok(self.register_pipeline_raw_input(&pipeline, input_id)?))
+                    .collect::<Result<Vec<_>>>()?;
+                thread::spawn(move || raw_data_sender(frame_senders, input));
+            }
+            InputFile::Mp4(path) => {
+                for input_id in &inputs {
+                    self.register_pipeline_mp4_input(&pipeline, input_id, &path)?;
+                }
+            }
+        };
+
+        let scene_ctx = SceneContext {
+            inputs: inputs.clone(),
+            outputs: outputs.clone(),
+        };
+        let receivers = outputs
+            .iter()
+            .map(|output_id| {
+                let root = (self.scene_builder)(&scene_ctx, output_id);
+                let receiver: Box<dyn DurationReceiver + Send> = match self.encoder {
+                    EncoderOptions::Disabled => {
+                        self.register_pipeline_raw_output(&pipeline, output_id, root)?
+                    }
+                    EncoderOptions::Enabled(preset) => {
+                        self.register_pipeline_encoded_output(&pipeline, output_id, root, preset)?
+                    }
+                };
+
+                Ok(receiver)
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let start_time = Instant::now();
+        let output_threads: Vec<_> = receivers
+            .into_iter()
+            .enumerate()
+            .map(|(index, receiver)| {
+                let warm_up_time = self.warm_up_time;
+                let measure_time = self.measure_time;
+                let error_tolerance_multiplier = self.error_tolerance_multiplier;
+                debug!("start listening for output frames");
+                thread::spawn(move || {
+                    debug!(?index, "start drain in warm up mode");
+                    while start_time.elapsed() < warm_up_time {
+                        if let Err(TryRecvError::Empty) = receiver.try_receive() {
+                            thread::sleep(Duration::from_millis(1));
+                        }
+                    }
+
+                    debug!(?index, "start drain in measure mode");
+                    let mut max_pts: Duration = Duration::ZERO;
```

Comment:
it needs to be based on duration, previous version counted frames which is correct for raw input, but not for encoded one

Reply:
I don't understand why.

Reply:
if we are getting encoded chunks we don't know how many frames there are

Reply:
oh, obviously. I've been sitting in our encoder for so long I forgot those can be split into chunks